revocation/ocsp/ocsp.go

```diff
@@ -0,0 +1,259 @@
+// Package OCSP provides methods for checking the OCSP revocation status of a
+// certificate chain, as well as errors related to these checks
+package ocsp
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/notaryproject/notation-core-go/revocation/base"
+	coreX509 "github.com/notaryproject/notation-core-go/x509"
+	"golang.org/x/crypto/ocsp"
+)
+
+// Options specifies values that are needed to check OCSP revocation
+type Options struct {
+	CertChain   []*x509.Certificate
+	SigningTime time.Time
+	HTTPClient  *http.Client
+}
+
+const (
+	pkixNoCheckOID      string = "1.3.6.1.5.5.7.48.1.5"
+	ocspMaxResponseSize int64  = 20480 //bytes
+)
+
+// CheckStatus checks OCSP based on the passed options and returns an array of
+// base.CertRevocationResult objects that contains the results and error. The
+// length of this array will always be equal to the length of the certificate
+// chain.
+func CheckStatus(opts Options) ([]*base.CertRevocationResult, error) {
+	result := make([]*base.CertRevocationResult, len(opts.CertChain))
+
+	if len(opts.CertChain) == 0 {
+		return result, nil
+	}
+
+	// Validate cert chain structure
+	if err := coreX509.ValidateCodeSigningCertChain(opts.CertChain, &opts.SigningTime); err != nil {
+		return nil, base.InvalidChainError{Err: err}
+	}
```

Comment:
I agree with the intention of moving it into validate. However, since users can potentially use ocsp.CheckStatus directly, I think it is not the best approach. Otherwise, a user could call this function directly with an invalid chain and it would potentially return odd errors due to performing OCSP checks for the wrong certs rather than failing quickly.